### ISSUE TITLE
[EPUB 3.3] [suggestion] Note for non-complete Media Overlay

### ISSUE THREAD

**Issue by gregoriopellegrino** [w3c/epub-specs#1738]
The specification says:

> "EPUB Creators place par elements together sequentially to form a series of phrases or sentences. Not every element of the EPUB Content Document will have a corresponding par element in the Media Overlay, only those relevant to the audio narration.".
> https://www.w3.org/TR/epub-33/#sec-docs-intro

Perhaps insert a note indicating that for avoiding accessibility issues it would be good if all textual content is also represented in audio, or at least alert the user to the presence of "extra" content?

**Comment by mattgarrish** [comment]
I believe the intent here is only that the smil file won't exactly match every styling/presentational element used in the content document.

But it's an interesting question whether you have to provide a complete rendering of the text content in media overlays to conform to the accessibility standard. We don't require that currently, either within a file or across the entire publication.

For example, what if you don't narrate an index? Are we ready to make that a failure? What if there were repetitive text content that wasn't necessary to narrate?

I'm not sure where I stand on this, to be honest.

**Comment by avneeshsingh** [comment]
We should see Media Overlays as replacement of screen reader or ReadAloud feature. 
The main purpose of full text full audio approach (termed as Media Overlays in EPUB 3) is to overcome the limitation of languages which do not have good TTS + it is very useful for people who find it difficult to perceive synthetic speech for example learning disabilities, hearing impaired etc.
So, for accessibility perspective, what ever text is supposed to be available to screen reader or ReadAloud, it should be recorded in Media Overlays also. Which means all the meaningful text should be available in audio.

Coming to the specifications. We have separate specification for accessibility of EPUB. So, e can allow Media Overlays specs to provide some flexibility because some publishers may be using Media Overlays for reasons other than accessibility. And address this issue in Media Overlays section of EPUB accessibility specifications.

**Comment by gregoriopellegrino** [comment]
My "concern" is about the [FAST Checklist](https://w3c.github.io/apa/fast/checklist.html) points:

> # If technology allows text content
> ## Authors can define non-text alternatives for text content.
> While text is the universal accessible alternative, it is still not the best format for some users. Technologies should allow authors to provide non-text alternatives to text content when needed. Various types of alternatives are useful in different situations, including visual such as icons or movies, auditory such as pronunciation cues and recorded speech, and haptic.
> ## Authors can define non-text alternatives for non-text content.
> Even though text alternatives for non-text content is generally recommended, in some situations a non-text alternative is more suitable. For instance, a haptic version of a map, in which features are conveyed by touch features, can be easier to understand than a text alternative. Technologies should allow authors to provide these enhance alternatives in addition to text alternatives.

But it actually doesn't talk about having **all** the content read by the speaker, just offer the content creator the ability to do so.

**Comment by avneeshsingh** [comment]
@gregoriopellegrino wrote:
"But it actually doesn't talk about having all the content read by the speaker, just offer the content creator the ability to do so."


This is completely aligned to the approach that I explained. As far as the basic concepts of web accessibility are concerned, the existing Media Overlays specifications will pass it. The a11y horizontal review does not consider the special cases like Media Overlays. It is based on web content in general. APA did work on a document on media, but it is more of a note at this stage. It future it may be included in fast check list.

**Comment by avneeshsingh** [comment]
My second comment on updating Media Overlays section in EPUB Accessibility goes beyond a11y horizontal review.
Lets recall the long discussion we had while creating accessModeSufficient. For full text full audio or audio sync text, the meaningful content should be available in:
[auditory]
[textual]
[auditory, textual]

So, a line for clarifying this requirement in MO section of EPUB Accessibility should be sufficient.

**Comment by mattgarrish** [comment]
> So, a line for clarifying this requirement in MO section of EPUB Accessibility should be sufficient.

What are you proposing here, @avneeshsingh ?

Do you want to add a new requirement that media overlays must or should include audio for all text content?

**Comment by avneeshsingh** [comment]
Yes, this clarification should be provided in Media Overlays section of EPUB accessibility.
All the meaningful content of An accessible Media Overlays publication should be read through the following modalities:
Textual
Auditory
Textual auditory

So, we should add a note that accessible Media Overlays should provide audio equivalent for all the meaningful content of the publication. (I have intentionally used audio equivalent instead of audio recording)
Using "must" vs "should" needs more discussion.

**Comment by mattgarrish** [comment]
> Using "must" vs "should" needs more discussion.

Right, it's just that our objectives are currently all informative even though we define them in normative terms (we probably should have an informative label on the objectives section to make this extra clear). The only change we made so far this revision is to require authors to conform to the media overlays specification.

That's why even if we frame this as a recommendation or requirement in an objective rather than a note, it would still only be informative. It would also give us more room to explain the issue, as I don't believe we can rely on schema.org terms.

What if I craft a "requirement" and then we can pick this up when calls resume?

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-08-12-epub-a11y#section1) on 2021-08-12

List of resolutions:

- ***[Resolution No. 1](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-08-12-epub-a11y#resolution1):  approve issue #1738***

<details><summary><i>View the transcript</i></summary>

### 1. Pull request for Media Overlays section: all meaningful text should also have audio.

_See github pull request [#1759](https://github.com/w3c/epub-specs/pull/1759)._




_See github issue [#1738](https://github.com/w3c/epub-specs/issues/1738)._




**Avneesh Singh:** the background is that in MO there was no specific requirement that you should have audio for every meaningful text element  
… for a11y this is very important because MO provides experiences like screen reader  
… the PR above addresses this need  

**George Kerscher:** whenever I play MO I start it and let it go. I expect that everything that should be presented to me is presented. I'd never think to go back and look for content that is skipped. So this change seems perfect.  
… if there is a portion of a book that is presented as MO, then that portion should be self-contained  

**Avneesh Singh:** these requirements are informative so as not to break backwards compatibility  
… we are informing publishers that this is our expectation  
… in the future we may make this normative (i.e. future version of spec)  

> **Proposed resolution: approve issue #1738** *(Avneesh Singh)*

> *George Kerscher:* +1

> *Will Awad:* +1

> *Matthew Chan:* +1

> *Murata Makoto:* +1

> *Vijaya Gwori Perumal:* +1

> ***Resolution #1: approve issue #1738***

</details>
